client/components/keyed-suggestions/index.jsx

```diff
@@ -445,14 +458,24 @@ class KeyedSuggestions extends Component {
 					</span>
 					{ Object.keys( this.props.terms[ key ] ).length > suggestions[ key ].length && (
 						<SuggestionsButtonAll
-							onClick={ this.onShowAllClick }
+							onClick={ ( category ) => {
+								this.onShowAllClick( category );
+								this.props.recordTracksEvent( 'search_dropdown_show_all_button_click', {
```

Comment:
Suuuuuuper nit, maybe no action needed: note that the button actually says `View all`, not `Show all` 😂